Add ansi-enable and ansi-disable commands

SeerLite opened the pull request:
To be used in custom hooks in user configuration:
```kak
hook global BufCreate '\*make\*' ansi-enable
```

krobelus commented on rc/ansi.kak:
```diff
@@ -67,7 +67,11 @@ define-command \
+    -docstring %{ansi-enable: render buffer and enable rendering new fifo data for current buffer.} \
+    -params 0 \
+    ansi-enable %{
+    ansi-render
```
I like this because up to now I have been too lazy to write this myself. 

The lack of an `ansi-disable` command seems acceptable since this is mostly used for `*make*` - the user can just create a new `*make*` buffer instead.

SeerLite replied:
Yeah but now that you point it out, maybe I really should add an `ansi-disable`, for completeness sake.

Actually now that I think about it I can't see many uses for the `ansi-render` and `ansi-clear` commands alone. `ansi-enable` and `ansi-disable` could do the same thing without causing any conflicts, with the added benefit that they work on fifos too.

Maybe we should consider replacing the original commands altogether?

krobelus replied:
`ansi-enable; ansi-render` seems usually better than just `ansi-render`. However, on a very large buffer, `ansi-render` can take a considerable amount of time. So maybe there is still merit in doing it manually,like your current version does. Probably not a strong use case though; I think either way would work.

krobelus commented on rc/ansi.kak:
```diff
@@ -67,7 +67,11 @@ define-command \
-hook -group ansi global BufCreate '\*stdin(?:-\d+)?\*' %{
+define-command \
+    -docstring %{ansi-enable: render buffer and enable rendering new fifo data for current buffer.} \
+    -params 0 \
```
I think these `-params 0` lines should be removed because it's the default, but that doesn't have to be in this PR.

eraserhd replied:
I actually like the explicitness.

krobelus reviewed (commented):
Readme needs updating as well.

SeerLite commented:
I updated the README and also added `ansi-disable`.

@eraserhd What do you think about replacing render/clear commands with enable/disable commands? I can't see many uses for only render/clear but maybe I'm missing something

eraserhd reviewed (approved):
This is nice, thanks!

eraserhd commented:
Regarding replacing render/clear with enable/disable... I'm not really sure.

I'm all for a simpler interface.  Assuming ansi-enable renders the buffer and ansi-disable clears the highlighting for the buffer, here are a few thoughts:

* I'm ok with adding fifo hooks to non-fifo buffers.
* Do we ever want to disable then re-enable on a buffer?  That would clear existing colors.

Ok, so I think I'd propose:

* `ansi-enable` calls `ansi-render` and adds hooks, so it works on static buffers too.
* `ansi-disable` removes hooks, does not clear colors.
* `ansi-clear` clears colors.
* We recommend people normally use `ansi-enable`.
